### PR TITLE
feat: idp v2 api GetIDPByID

### LINKED ISSUE #8337: Provide `getIDPById` method
### Preflight Checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions
- [ ] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

When implementing the login/register with IDP, after completing an IDP flow, we have to get information about the IDP to decide about the next steps like:

- user link exists -> login
- user link does not exist -> register (`createUser`)
- user link does not exist, but user with same username / email exists -> link User (`addIDPLink`)

### Describe your ideal solution

The Response of the `getIDPById` must include the following information
``` 
is_linking_allowed: bool
is_creation_allowed: bool
is_auto_creation: bool
is_auto_update: bool
auto_linking: AutoLinkingOption
```

### Version

_No response_

### Environment

ZITADEL Cloud

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by stebenz**
# Which Problems Are Solved

GetIDPByID as endpoint in the API v2 so that it can be available for the new login.

# How the Problems Are Solved

Create GetIDPByID endpoint with IDP v2 API, throught the GetProviderByID implementation from admin and management API.

# Additional Changes

- Remove the OwnerType attribute from the response, as the information is available through the resourceOwner.
- correct refs to messages in proto which are used for doc generation
- renaming of elements for API v3

# Additional Context

Closes #8337

### CODE REVIEW

File: proto/zitadel/idp/v2/idp_service.proto [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      }
+    }
+  }
+};
+
+service IDPService {
+
+  // Get Identity Provider (IDP) by ID
+  //
+  // Returns an identity provider (social/enterprise login) by its ID e.g Google, AzureAD, etc.
```
[the remark this answers is not in the document]
Comment (stebenz) [reply proto/zitadel/idp/v2/idp_service.proto:107]: Used the same description as in mgmt and admin for less confusion, can try to make it more self-explanatory.

File: proto/zitadel/idp/v2/idp.proto
[the remark this answers is not in the document]
Comment (stebenz) [reply proto/zitadel/idp/v2/idp.proto:1]: In mgmt and admin the resources are called provider as we couldn't use IDP, and are unter '/idps/templates', which are a strange combination.
Used 'IDP' as defined in the V3.

File: proto/zitadel/idp/v2/idp_service.proto
[the remark this answers is not in the document]
Comment (stebenz) [reply proto/zitadel/idp/v2/idp_service.proto:1]: That's true that could be used for the naming of the service, would make more sense to not shorten it.

File: internal/api/grpc/idp/v2/query.go [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+		}
+	}
+	return &idp_pb.GetIDPByIDResponse{Idp: idpToPb(idp)}, nil
+}
+
+func idpToPb(idp *query.IDPTemplate) *idp_pb.IDP {
+	changeDate := idp.CreationDate
+	if !idp.ChangeDate.IsZero() {
+		changeDate = idp.ChangeDate
+	}
```
[the remark this answers is not in the document]
Comment (stebenz) [reply internal/api/grpc/idp/v2/query.go]: removed

File: internal/api/grpc/idp/v2/query.go [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+			}),
+		State:  idpStateToPb(idp.State),
+		Name:   idp.Name,
+		Type:   idpTypeToPb(idp.Type),
+		Config: configToPb(idp),
+	}
+}
+
+func idpStateToPb(state domain.IDPState) idp_pb.IDPState {
+	switch state { //nolint:exhaustive
```
[the remark this answers is not in the document]
Comment (stebenz) [reply internal/api/grpc/idp/v2/query.go]: removed and added state in proto

File: internal/api/grpc/idp/v2/query.go [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+	switch idp.OwnerType {
+	case domain.IdentityProviderTypeSystem:
+		if err := s.checkPermission(ctx, domain.PermissionIDPRead, idp.ResourceOwner, idp.ID); err != nil {
+			return nil, err
+		}
+	case domain.IdentityProviderTypeOrg:
+		if err := s.checkPermission(ctx, domain.PermissionOrgIDPRead, idp.ResourceOwner, idp.ID); err != nil {
+			return nil, err
+		}
+	}
```
[the remark this answers is not in the document]
Comment (stebenz) [reply internal/api/grpc/idp/v2/query.go:33]: As the query side functions are used in different APIs and the permission check is already done in the API for example for the users, should we rework that as well to keep it consistent?

**Review by livio-a — APPROVED** [review]

File: proto/zitadel/resources/action/v3alpha/action_service.proto [hunk 5]
```diff
@@ -135,7 +135,7 @@ service ZITADELActions {
           description: "Target successfully created";
           schema: {
             json_schema: {
-              ref: "#/definitions/CreateTargetResponse";
+              ref: "#/definitions/v3alphaCreateTargetResponse";
```
Comment (eliobischof) [inline proto/zitadel/resources/action/v3alpha/action_service.proto:138]: is this change intended?

Comment (stebenz) [reply proto/zitadel/resources/action/v3alpha/action_service.proto:138]: Yes, the generate throws errors otherwise, was v2CreateTargetResponse before, which worked, then was CreateTargetResponse, so I changed it to v3alpha...

Comment (eliobischof) [reply proto/zitadel/resources/action/v3alpha/action_service.proto:138]: > correct refs to messages in proto which are used for doc generation

Does this comment refer to this change?
What does that do? I'm just curious.

Comment (stebenz) [reply proto/zitadel/resources/action/v3alpha/action_service.proto:138]: image

Comment (stebenz) [reply proto/zitadel/resources/action/v3alpha/action_service.proto:138]: Thats for the status code 201, otherwise it would only be done generated in the swagger for 200.

Comment (eliobischof) [reply proto/zitadel/resources/action/v3alpha/action_service.proto:138]: thanks

**Review by eliobischof — DISMISSED** [review]

**Review by eliobischof — COMMENTED** [review]
image

**Review by eliobischof — APPROVED** [review]
